python/lsst/ip/isr/isrTask.py

```diff
@@ -392,11 +402,13 @@ def readIsrData(self, dataRef, rawExposure):
         """
         ccd = rawExposure.getDetector()
 
-        biasExposure = self.getIsrExposure(dataRef, "bias") if self.config.doBias else None
+        biasExposure = self.getIsrExposure(dataRef, self.config.biasDataProductName) \
+            if self.config.doBias else None
         # immediate=True required for functors and linearizers are functors; see ticket DM-6515
         linearizer = dataRef.get("linearizer", immediate=True) if self.doLinearize(ccd) else None
         darkExposure = self.getIsrExposure(dataRef, "dark") if self.config.doDark else None
-        flatExposure = self.getIsrExposure(dataRef, "flat") if self.config.doFlat else None
+        flatExposure = self.getIsrExposure(dataRef, self.config.flatDataProductName) \
+            if self.config.doFlat else None
```

Comment:
I'm confused why this is in a ticket for adding crosstalk.  It seems to be complicating the issue.

Also, it seems odd to only generalize two of them.  Why not also generalize `dark`?

Reply:
This came up when I was writing a test which calls DecamCpIsrTask, which had its own copy-pasted readIsrData method for the specially-named cpBias and cpFlat issue alone. It was slowly diverging from ip_isr's readIsrData which is what I wanted to use anyway, so instead of perpetuating the copy-paste cycle I thought it best to just add this fix. (There aren't such a thing as cpDarks so I let it be.)

Reply:
I see.  Please add the generalization for `dark`.  I think it would be pretty confusing to someone who uses this to have access to only two of the three.